src/Microsoft.AspNetCore.StaticFiles/StaticFileOptions.cs

```diff
@@ -41,6 +42,17 @@ public StaticFileOptions(SharedOptions sharedOptions) : base(sharedOptions)
         public string DefaultContentType { get; set; }
 
         /// <summary>
+        /// The default cache profile settings to apply to all static files served. 
+        /// </summary>
+        public CacheProfile DefaultCacheProfile { get; set; }
+
+        /// <summary>
+        /// Called to provide a specific <see cref="CacheProfile"/> for the given 
+        /// <see cref="StaticFileResponseContext"/>. Return null to use the <see cref="DefaultCacheProfile"/>.
+        /// </summary>
+        public Func<StaticFileResponseContext, CacheProfile> CacheProfileProvider { get; set; }
```

Comment:
We probably need to extract this into a service to allow for DI when creating the cache profile.